### PR TITLE
Added conanfile.py

### PR DESCRIPTION
**Pull request by zeokav**
Added conan configuration file that builds a package containing rang.hpp in include. No files have been exported and sources are downloaded from the origin git repo. A conan.io maintainer has uploaded it to conan under the package name - rang/1.0@memsharded/stable

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    Build policy set to missing to avoid the --build flag since only a header is required.
+    '''
+    name = "rang"
+    version = "1.0"
+    url = "https://github.com/agauniyal/rang.git"
+    build_policy = "missing"
+
+    def source(self):
+        self.run("git clone https://github.com/agauniyal/rang.git")
+        self.run("cd rang")
```
Comment (agauniyal) [inline conanfile.py]: should also checkout some git tag eg `git checkout v1.0`

**Comment by agauniyal** [comment]
Also what would be the workflow here? Say if I've to include rang lib in another `hello.cpp` project of mine, how would I proceed..

**Comment by zeokav** [comment]
About the workflow, I finally understood how this works so I'll be happy to explain. :smile: 
So how conan works is, when you need to update the remotely stored package, all you need to do is change version in the conanfile.py provided in the project folder (and obviously if there are any other changes to it), and from the project directory do a `$conan export zeokav/stable ` where zeokav is my username and stable is my 'channel' (Like different branches). 
What this does is, stores a copy of the code locally, which can then be shared by other projects on the same system. To upload this to the conan.io database, you need to `conan upload <remote>`, and it gets updated in the remote directory.

To use it in any other project, all you need to do is make a conanfile.txt that describes the requirements of the project. So in our case, the conanfile would be something like - 

> [requires]
> rang/1.0@zeokav/stable

> [imports]
> include, *.hpp -> ../include

This followed by `$mkdir build && cd build; conan install ..`
What this does is, from the package rang, v1.0 under zeokav/stable, it copies the .hpp files in the include folder of the package to the include directory of the project you're working on. After that, it's a simple `#include "include/rang.hpp"`

I hope this makes things clear. :smile:

**Comment by agauniyal** [comment]
Could you remove the cache file, you must've accidentally committed it - `__pycache__/conanfile.cpython-35.pyc`. I'll merge it as soon as you're done 👍 Thanks for the PR 😄

**Comment by zeokav** [comment]
Oh yes. I forgot about that. Done! 👍

**Comment by agauniyal** [comment]
Aaaaaand it is done 🎉

**Pull request merged**